File: cadwiki-nuget/cadwiki.AutoCAD2021.Base.Utilities/Workflows/NunitTestRunner.vb
Option Strict On
Option Infer Off
Option Explicit On

Imports Autodesk.AutoCAD.ApplicationServices
Imports Autodesk.AutoCAD.EditorInput
Imports cadwiki.NUnitTestRunner.Results
Imports cadwiki.NUnitTestRunner.UI

Namespace Workflows
    Public Class NunitTestRunner
        Public Async Function Run(ByVal regressionTestTypes As Type()) As Task
            Dim doc As Document = Core.Application.DocumentManager.MdiActiveDocument
            Dim ed As Editor = doc.Editor
            Try
                Dim results As ObservableTestSuiteResults = New cadwiki.NUnitTestRunner.Results.ObservableTestSuiteResults()
                Dim driver As WpfDriver = New cadwiki.NUnitTestRunner.UI.WpfDriver(results, regressionTestTypes)
                Dim window As WindowTestRunner = driver.GetWindow()
                ' https://forums.autodesk.com/t5/net/how-to-set-a-focus-to-autocad-main-window-from-my-form-of-c-net/td-p/4680059
                Core.Application.ShowModelessWindow(window)
                Await driver.ExecuteTestsAsync()
            Catch ex As Exception
                ExceptionHandler.Handle(ex)
            End Try
        End Function
    End Class
End Namespace

File: cadwiki-nuget/cadwiki.AC/Workflows/NunitTestRunner.cs
using System;
using System.Threading.Tasks;
using cadwiki.NUnitTestRunner.Results;

namespace cadwiki.AC.Workflows
{
    public class NunitTestRunner
    {
        public async Task Run(Type[] regressionTestTypes)
        {
            var doc = global::Autodesk.AutoCAD.ApplicationServices.Core.Application.DocumentManager.MdiActiveDocument;
            var ed = doc.Editor;
            try
            {
                var results = new ObservableTestSuiteResults();
                var driver = new NUnitTestRunner.Ui.WpfDriver(ref results, regressionTestTypes);
                var window = driver.GetWindow();
                // https://forums.autodesk.com/t5/net/how-to-set-a-focus-to-autocad-main-window-from-my-form-of-c-net/td-p/4680059
                global::Autodesk.AutoCAD.ApplicationServices.Core.Application.ShowModelessWindow(window);
                await driver.ExecuteTestsAsync();
            }
            catch (Exception ex)
            {
                ExceptionHandler.WriteToEditor(ex);
            }
        }
    }
}